### ISSUE TITLE
Optimizers are not compatible with all Layer Types

### ISSUE THREAD

**Issue by StuartFarmer** [N3PDF/evolutionary_keras#36]
Optimizers should support all layer types to be a true drop-in replacement for other Keras optimizers.

**Comment by scarlehoff** [comment]
You are right, sadly we can't realistically keep pace with Keras so the scope has narrowed to being drop-in for our subsets of needs only :(
